test/videojs-http-streaming.test.js

```diff
@@ -1160,6 +1160,99 @@ QUnit.test(
   }
 );
 
+QUnit.test(
+  'systemBandwidth retrieves bandwidth from networkInformation when option is enabled',
+  function(assert) {
+    this.player = createPlayer({ html5: { vhs: { useNetworkInformation: true } } });
+    this.player.src({
+      src: 'manifest/master.m3u8',
+      type: 'application/vnd.apple.mpegurl'
+    });
+
+    this.clock.tick(1);
+
+    this.player.tech_.vhs.bandwidth = 10e10;
+    this.player.tech_.vhs.effectiveBandwidthInBits_ = 20e10;
+    this.player.tech_.vhs.throughput = 20e10;
+    // 1 / ( 1 / 20e10 + 1 / 20e10) = 10e10
+    assert.strictEqual(
+      this.player.tech_.vhs.systemBandwidth,
+      10e10,
+      'systemBandwidth is the combination of networkInfo.downlink and throughput'
+    );
+  }
+);
+
+QUnit.test(
+  'systemBandwidth uses player-estimated bandwidth when its value is greater than networkInformation.downLink',
+  function(assert) {
+    this.player = createPlayer({ html5: { vhs: { useNetworkInformation: true } } });
+    this.player.src({
+      src: 'manifest/master.m3u8',
+      type: 'application/vnd.apple.mpegurl'
+    });
+
+    this.clock.tick(1);
+
+    this.player.tech_.vhs.bandwidth = 20e10;
+    this.player.tech_.vhs.effectiveBandwidthInBits_ = 10e10;
+    this.player.tech_.vhs.throughput = 20e10;
+    // 1 / ( 1 / 20e10 + 1 / 20e10) = 10e10
+    assert.strictEqual(
+      this.player.tech_.vhs.systemBandwidth,
+      10e10,
+      'systemBandwidth is the combination of player bandwidth and throughput'
+    );
+  }
+);
+
+QUnit.test(
+  'effectiveBandwidthInBits_ default',
+  function(assert) {
+    this.player.src({
+      src: 'manifest/master.m3u8',
+      type: 'application/vnd.apple.mpegurl'
+    });
+
+    this.clock.tick(1);
+
+    assert.strictEqual(
+      this.player.tech_.vhs.effectiveBandwidthInBits_,
+      0,
+      'effectiveBandwidthInBits_ defaults to 0'
```

Comment:
Would it be worth having a default value? Or reading the initial value?

Reply:
downlink will provide a default value. From the docs: "In the absence of recent bandwidth measurement data, the attribute value is determined by the properties of the underlying connection technology." Given this, I don't think there is a need for us to test for the default. Also, we'll always choose the larger value between Vhs' bandwidth estimate and the estimate provided by downlink, so we should be safe even if downlink estimates a low initial value for some reason